policy-controller/src/admission.rs

```diff
@@ -321,7 +321,9 @@ impl Validate<ServerSpec> for Admission {
                 && server.spec.port == spec.port
                 && Self::overlaps(&server.spec.pod_selector, &spec.pod_selector)
             {
-                bail!("identical server spec already exists");
+                let existing = serde_yaml::to_string(&server.spec)
+                    .unwrap_or_else(|err| format!("error serializing yaml: {err}"));
+                bail!("identical server spec '{ns}/{name}' already exists:\n{existing}",);
```

Comment:
this should be `{server.name_unchecked}` I think, to be the name of the *existing* server that this new one conflicts with.